components/UI/FilterMenu/TileLayer.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState, MouseEvent } from "react";
+import { Menu, MenuItem } from "@mui/material";
+import FilterMenuButton from "./FilterMenuButton";
+import { useTranslation } from "react-i18next";
+
+export const tileLayerMenuOptions: readonly TileLayerMenuOption[] = [
+  {
+    label: "map",
+    value: "m",
+  },
+  {
+    label: "satellite",
+    value: "s",
+  },
+] as const;
+
+export const initialTileLayer = tileLayerMenuOptions[0];
+
+export interface TileLayerMenuOption {
+  label: string;
+  value: string;
+}
+
+const valueToOption = (value: string): TileLayerMenuOption => {
+  return tileLayerMenuOptions.find(
+    (option) => option.value === value
+  ) as TileLayerMenuOption;
+};
+
+export interface TileLayerMenuProps {
+  onChange: (_option: TileLayerMenuOption) => void;
+}
+
+const TileLayerMenu: React.FC<TileLayerMenuProps> = ({ onChange }) => {
```

Comment:
This component has a lot of shared logic with the other buttons in the filter menu, we should ideally refactor this.